pkg/ingestor/parser/csaf/parser_csaf.go

```diff
@@ -146,7 +151,7 @@ func (c *csafParser) findPkgSpec(ctx context.Context, product_id string) (*gener
 		return nil, fmt.Errorf("unable to locate product reference for id %s", product_id)
 	}
 
-	purl := findPurl(ctx, c.csaf.ProductTree, *pref)
+	purl := findPurl(ctx, c.csaf.ProductTree, *pref, make(map[string]bool))
```

Comment:
can we have this be the same and just have it call the recursive function, so rename `findPurl` to `findPurlSearch` and then
```
func findPurl(ctx, c.csaf.ProductTree, *pref) {
return findPurlSearch(ctx, c.csaf.ProductTree, *pref, make(map[string]bool))
}

Reply:
Good idea, done!